packages/perseus/src/widgets/__tests__/interactive-graph.test.ts

```diff
@@ -104,3 +111,70 @@ describe("interactive-graph widget", function () {
         },
     );
 });
+
+describe("segment graph", () => {
+    let userEvent: UserEvent;
+    beforeEach(() => {
+        userEvent = userEventLib.setup({
+            advanceTimers: jest.advanceTimersByTime,
+        });
+    });
+
+    const apiOptions = {flags: {mafs: {segment: true}}};
+
+    it("should render", () => {
+        renderQuestion(segmentQuestion, apiOptions);
+    });
+
+    it("should reject when has not been interacteracted with", () => {
+        // Arrange
+        const {renderer} = renderQuestion(segmentQuestion, apiOptions);
+
+        // Act
+        // no action
+
+        // Assert
+        expect(renderer).toHaveInvalidInput();
+    });
+
+    it("rejects incorrect answer", async () => {
+        // Arrange
+        const {renderer, container} = renderQuestion(
+            segmentQuestion,
+            apiOptions,
+        );
+
+        // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
+        const movablePoints = container.querySelectorAll(
+            "circle.mafs-movable-point-hitbox",
+        );
+
+        // Act
+        await userEvent.type(movablePoints[0], "{arrowup}{arrowdown}");
```

Comment:
Really really glad we can drive these with keyboard events. It's a double-win in that we are testing keyboard support right out of the gate!  Nicely done!